### ISSUE TITLE
Different scores from JS library

### ISSUE THREAD

**Issue by feanz** [mickford/zxcvbn-cs#4]
![image](https://cloud.githubusercontent.com/assets/507875/18999021/3f20ad78-8733-11e6-8b53-1f051128b7a6.png)

Looks like I get different scores using zxcvbn-cs and zxcvbn.  Ideally I'd use both one for front end validation and another for server side.  It looks like this is happening due to the .net implementation score being based of of crack time not total guesses.

**Comment by afomichenko** [comment]
In our testing we came across a password to which JS gives a score of 3, and C# gives a score of 1.

Sample password: Sweetpea20151970

This renders it utterly unreliable to validate the password on both, client side and server side.

**Comment by UnderTheGun** [comment]
This is a very large issue and we beg the community to resolve it ASAP.  We get nearly 2,500 support center calls per month regarding password resets, and implementing these two libraries has caused much greater user frustration.  Users enter their password into the UI and it validates successfully (score 3), but then the server component rejects it (score 1 or 2).  To get around this we had to artificially inflate the C# score by 1, but this is both not advisable and not a very effective solution.  We still get tons of frustrated users who successfully choose a password which is then rejected on the server side.  We hope this will be resolved soon.

**Comment by JLeeC** [comment]
Agreed with afomichenko.  Server side should of course never expect input to be correct, so it makes sense for both the UI and server to perform the check.  Since this is a port from the js library, the algorithm should yield the same results and should be idempotent on both UI and server.

**Comment by adamcaudill** [comment]
It is worth remembering that this library is based on the JS from roughly 4 years ago - much has changed with the JS in those 4 years, which isn't reflected here. 

I had looked into getting a PR together a year or so ago to bring it up to date, but determined I didn't have time to make all the changes. What's really needed is an automated approach so that it can easily be updated as each new release comes out - if you want the results to match.

**Comment by ashwinkumar6** [comment]
I was just wondering if there is an ETC for this issue

**Comment by odumetz** [comment]
We solved this by using https://github.com/Microsoft/ClearScript (avail as nuget), loading the minified zxcvbn js source in the script engine and wrapping into a method to get a score. works great.
Obviously results are now 100% the same.

**Comment by graingert** [comment]
has anyone seen this fork? https://github.com/trichards57/zxcvbn-cs

**Comment by mooseburger** [comment]
@graingert Tested the nuget package (https://www.nuget.org/packages/zxcvbn-core/), still has a score mismatch.

**Comment by mooseburger** [comment]
@graingert This one actually does match dropbox's implementation:

https://github.com/hexasoftuk/Hexasoft.Zxcvbn

**Comment by graingert** [comment]
@mooseburger perhaps using the Rust version compiled to WASM would be the best bet for support in browsers and .NET
